storage_broker/src/lib.rs

```diff
@@ -123,23 +124,13 @@ where
         }
     }
 
-    fn poll_data(
+    fn poll_frame(
         self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-    ) -> Poll<Option<Result<Self::Data, Self::Error>>> {
+    ) -> Poll<Option<Result<hyper_1::body::Frame<Self::Data>, Self::Error>>> {
         match self.get_mut() {
-            EitherBody::Left(b) => Pin::new(b).poll_data(cx).map(map_option_err),
-            EitherBody::Right(b) => Pin::new(b).poll_data(cx).map(map_option_err),
-        }
-    }
-
-    fn poll_trailers(
-        self: Pin<&mut Self>,
-        cx: &mut Context<'_>,
-    ) -> Poll<Result<Option<hyper::HeaderMap>, Self::Error>> {
-        match self.get_mut() {
-            EitherBody::Left(b) => Pin::new(b).poll_trailers(cx).map_err(Into::into),
-            EitherBody::Right(b) => Pin::new(b).poll_trailers(cx).map_err(Into::into),
+            EitherBody::Left(b) => Pin::new(b).poll_frame(cx).map(map_option_err),
+            EitherBody::Right(b) => Pin::new(b).poll_frame(cx).map(map_option_err),
```

Comment:
feels weird if http-body did not have this.

Reply:
oki it doesn't have, but https://docs.rs/http-body-util/latest/http_body_util/enum.Either.html would be it. though the http-body-util is only 0.1.2 version, but then again, might be easier to keep upgrading...?

Reply:
Using `Either` from that crate now (was already depending on it).